p2p/switch.go

```diff
@@ -637,10 +646,8 @@ func (sw *Switch) addPeer(p Peer) error {
 	p.SetLogger(sw.Logger.With("peer", p.NodeInfo().NetAddress()))
 
 	// All good. Start peer
-	if sw.IsRunning() {
```

Comment:
did not understand this. if the switch is not running, we don't start the peer, so... it stays there presumably forever.

Reply:
I think this is to handle the shut down case where the switch has stopped but we're concurrently trying to add a peer. You're right this might be a leak, but if we were shutting down anyways, it wouldn't be an issue. I'm afraid with out it this could cause a panic during the shutdown? Not sure

Reply:
ok. added a comment & log statement